src/js/BuildApi.js

```diff
@@ -149,6 +149,17 @@ export default class BuildApi {
         });
     }
 
+    requestBuildOptions(key, onSuccess, onFailure) {
+        const url = `${this._url}/api/builds/${key}/json`;
+        $.get(url, function (data) {
+            onSuccess(data);
+        }).fail(xhr => {
+            if (onFailure !== undefined) {
+                onFailure();
+            }
+        });
+    }
```

Comment:
Can we promisify this instead then just use `try/catch` or `then/catch`?

Reply:
We are getting the features from the cloud build information? This is not the way. What if the build is very old, we will store ALL the builds in the time? What if the build is a local custom build? If we want to get this information, I think we need to create MSP commands in the firmware, that can tell us what features/other UI elements are enabled or not.

Reply:
@chmelevskij was just following same pattern in the api.
@McGiverGim there is a msp call to retrieve the buildkey. With the buildkey we request data from the cloud api filling some MSP variables. We might be able to store the filter in the cloud api. We only can use this from semver 1.45.
Local builds will have no buildkey (need to check).
I'd like to lock some configuration fields with cloud parameters as to hide items that are not included in the cloud build but the user still was able to select. Thoughts?

Reply:
As I say, in my opinion, the configurator must work with customs builds too, but the change to cloud build is a very big step and the Configurator had left outdated showing features that does not exist.
Maybe with the new MSP call with CLI we can ask for features to the CLI, if it shows an error then does not exist.

Reply:
I think there's two different issues here.

The first is using the build key, the user can default their build options to that contained for that key. Key not being user specific but build specific. Without key (custom or key not found) the build options revert to default.

The second is what actual features in configurator are enabled. Eg don't show GPS tab if GPS not built in. That should come from the firmware. Using the cli suggestion is good.

Reply:
@blckmn sorted now:

Buildkey availability determines behavior.
The tab issue is also resolved now.